File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dic = dict()
    
    for clo, typ in clothes:
        if typ not in dic:
            dic[typ] = 2
        else:
            dic[typ] += 1
    
    for i in dic.values():
        answer *= i
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> hm = new HashMap<>();
        
        for (String[] clo : clothes){
            hm.put(clo[1], hm.getOrDefault(clo[1], 1) + 1);
        }
        
        for (String key : hm.keySet()){
            answer *= hm.get(key);
        }
        
        return answer - 1;
    }
}